config/appConfig.json

```diff
@@ -0,0 +1,8 @@
+{
```

Comment:
@wwelling : I haven't done a full review of these changes, but wanted to leave an immediate comment.

I'm a little concerned about switching this configuration to a JSON config file, as JSON isn't the easiest to edit manually -- it's not very forgiving when you forget a trailing comma, it doesn't support comments, etc.

I'm also generally worried that copying default configs from `default-app-config.ts` over into this JSON file will be error prone, unless all users understand all the differences between JSON and Typescript formats.  This same issue would occur for people trying to migrate configs from the old `environment.prod.ts` format into this new JSON when upgrading from 7.1 to this PR.

Is there any way to possibly make this file Typescript?  Or somehow make migrations/upgrades easier?  I'm a bit worried that some users may stumble here during their next upgrade, and we need to keep in mind ways to simplify (or automate) the upgrade process as much as possible.

Reply:
I agree JSON is not the best application config format. I would suggest yaml or properties. I do not think typescript is an appropriate format for configuration. It is in Angular for static configuration and is the reason we all config determined at build time only.

I would not suggest copying configs from `default-app-config.ts`. It should only be used as reference due to lacking configuration documentation. I would suggest comprehensive documentation for the possible configurations as opposed to suggesting copying from anywhere.

For the configuration to be used at runtime it cannot be typescript. As it requires being hosted and/or serialized in the transfer state. 

It would be trivial to include a script to upgrade from a previous environment.ts file. Simply import and serialize as JSON. The script could take an argument to an existing environment.ts file.

If so desired to use a more common config format, I will gladly convert to YAML. Which would also be trivial to migrate. Import environment.ts file, serialize as JSON, and convert to YAML.

Reply:
@wwelling : Thanks for clarifying why you made this change. I think we should wait to hear from others as they review this PR.  My personal preference is YAML is easier to edit manually (and it supports inline comments, which I find useful in config files). But, I'd want to hear from others before we move from JSON to YAML.

I'd also want others opinions on the filename. While I'm not against `appConfig.json` or `appConfig.yml`, it might be easier/clearer if we just called it `config.json` or `config.yml`.

Reply:
@tdonohue sounds good to me. Neither change would be much more effort. I am currently adding some test coverage and type docs.

Reply:
YAML sounds good to me. I'd choose config.yml over appConfig.

Reply:
@artlowel @tdonohue I have converted the config file to TAML and refactored to `config.yml`. I have also added a script that can convert previous `environment.ts` files to `config.yml` files to simplify upgrading to this version. See update above.

Reply:
@tdonohue it appears I sort of contradicted myself above. It is possible to require or import typescript files at runtime and serializer them. As demonstrated in the env:yaml ts-node script. I am still not in favor of using typescript for runtime app configuration. It seems odd I guess.

We can put that back on the table and I can try it in this PR. Not entirely sure will work in node context after transpile.

Reply:
I'm not in favor of using *any* language powerful enough for general programming, as a configuration language.  Of the choices offered, let me add my support for YAML.

Reply:
@wwelling : Now that you have a migration script from `environment.*.ts` to `config.yml`, I'm OK with this configuration rename/reformatting change.  So, let's keep this PR at `config.yml` until other reviewers have a chance to take a look.  

Also, I highlighted this work in today's DSpace 7 DevMtg, and asked everyone to try to review it in the next week. Overall, there seems to be a lot of early support, but we need to get more eyes on the code to give you better feedback.  Thanks again for all your hard work on this!